src/Microsoft.AspNetCore.Hosting/WebHostBuilderExtensions.cs

```diff
@@ -72,5 +75,21 @@ public static IWebHostBuilder UseStartup(this IWebHostBuilder hostBuilder, Type
         {
             return hostBuilder.UseStartup(typeof(TStartup));
         }
+
+        /// <summary>
+        /// Configures the default service provider
+        /// </summary>
+        /// <param name="hostBuilder">The <see cref="IWebHostBuilder"/> to configure.</param>
+        /// <param name="configure">A callback used to configure the <see cref="ServiceProviderOptions"/> for the default <see cref="IServiceProvider"/>.</param>
+        /// <returns>The <see cref="IWebHostBuilder"/>.</returns>
+        public static IWebHostBuilder UseDefaultServiceProvider(this IWebHostBuilder hostBuilder, Action<ServiceProviderOptions> configure)
+        {
+            return hostBuilder.ConfigureServices(services =>
+            {
+                var options = new ServiceProviderOptions();
+                configure(options);
+                services.AddSingleton<IServiceProviderFactory<IServiceCollection>>(new DefaultServiceProviderFactory(options));
```

Comment:
Use `services.Replace` so you don't end up with two `IServiceProviderFactory<IServiceCollection>` in collection.